CrewChiefV4/Events/Strategy.cs

```diff
@@ -209,18 +214,49 @@ override protected void triggerInternal(GameStateData previousGameState, GameSta
             else if (currentGameState.SessionData.SessionType == SessionType.Race)
             {
                 // record the time each opponent entered the pitlane
-                foreach (OpponentData opponent in currentGameState.OpponentData.Values)
+                foreach (KeyValuePair<String, OpponentData> entry in currentGameState.OpponentData)
                 {
-                    if (opponent.InPits)
+                    if (entry.Value.IsNewLap)
                     {
-                        if (!opponentsInPitLane.ContainsKey(opponent.DriverRawName))
+                        Strategy.opponentsWhoWillExitCloseBehind.Remove(entry.Key);
+                        Strategy.opponentsWhoWillExitCloseInFront.Remove(entry.Key);
+                    }
+                    if (entry.Value.InPits)
+                    {
+                        if (!opponentsInPitLane.ContainsKey(entry.Value.DriverRawName))
                         {
-                            opponentsInPitLane.Add(opponent.DriverRawName, currentGameState.SessionData.SessionRunningTime);
+                            opponentsInPitLane.Add(entry.Value.DriverRawName, currentGameState.SessionData.SessionRunningTime);
+                            if (warnAboutOpponentsExitingCloseToPlayer && currentGameState.Now > nextOpponentPitExitWarningDue)
```

Comment:
why are we even entering this loop if feature is off?

Reply:
because we still collect up the opponentsInPitLane in this loop, which is used to adjust the expected opponent time loss estimate later